public/js/components/RightSidebar.js

```diff
@@ -22,73 +26,148 @@ function debugBtn(onClick, type, className, tooltip) {
-  return (
-    dom.div(
-      { className: "right-sidebar",
-        style: { overflowX: "hidden" }},
+    return pauseBtn(false, false, "Pause on all exceptions");
+  },
+
+  renderDisableBreakpoints() {
+    const { toggleAllBreakpoints, breakpoints,
+            breakpointsDisabled, breakpointsLoading } = this.props;
+
+    if (breakpoints.size == 0 || breakpointsLoading) {
+      return debugBtn(
+        null, "disableBreakpoints", "disabled", "Disable Breakpoints"
+      );
+    }
+
+    return debugBtn(
+      () => toggleAllBreakpoints(!breakpointsDisabled),
+      "disableBreakpoints",
+      breakpointsDisabled ? "breakpoints-disabled" : "",
+      "Disable Breakpoints"
+    );
+  },
+
+  getItems() {
+    const items = [
+    { header: "Breakpoints",
+      component: Breakpoints,
+      opened: true },
+    { header: "Call Stack",
+      component: Frames },
+    { header: "Scopes",
+      component: Scopes }
+    ];
+    if (isEnabled("watchExpressions")) {
+      items.unshift({ header: "Watch Expressions",
+        component: Expressions,
+        opened: true });
+    }
+    return items;
+  },
+
+  render() {
+    return (
       dom.div(
-        { className: "command-bar" },
-        pause ? [
-          debugBtn(() => command({ type: "resume" }), "resume",
-            "active", "Click to resume (F8)"),
-          debugBtn(() => command({ type: "stepOver" }), "stepOver",
-            "active", "Step Over (F10)"),
-          debugBtn(() => command({ type: "stepIn" }), "stepIn",
-            "active", "Step In (F11)"),
-          debugBtn(() => command({ type: "stepOut" }), "stepOut",
-            "active", "Step Out \u21E7 (F12)"),
-        ] : [
-          isWaitingOnBreak ?
-            debugBtn(null, "pause", "disabled", "Click to resume (F8)") :
-            debugBtn(breakOnNext, "pause", "active", "Click to resume (F8)"),
-          debugBtn(null, "stepOver", "disabled", "Step Over (F10)"),
-          debugBtn(null, "stepIn", "disabled", "Step In (F11)"),
-          debugBtn(null, "stepOut", "disabled", "Step Out \u21E7 (F12)")
-        ],
-
-        debugBtn(() => command({ type: "disableBreakpoints" }),
-                 "disableBreakpoints", "disabled", "Disable Breakpoints"),
-        debugBtn(() => pauseOnExceptions(!shouldPauseOnExceptions),
-                 "pause-exceptions",
-                 shouldPauseOnExceptions ? "enabled" : "disabled",
-          "Toggle Pause on Exceptions"),
-        debugBtn(() => command({ type: "subSettings" }), "subSettings",
-          "", "Settings")
-      ),
-
-      Accordion({
-        items: getItems()
-      })
-    )
-  );
-}
+        { className: "right-sidebar",
+          style: { overflowX: "hidden" }},
+        dom.div(
+          { className: "command-bar" },
+          this.renderStepButtons(),
+          this.renderDisableBreakpoints(),
+          this.renderPauseOnExceptions()
+        ),
+
+        Accordion({
+          items: this.getItems()
```

Comment:
Yea, I agree. If anything, I'd prefer if it were `renderAccordion` and then was grouped there.